File: Chapter05/5.4 Server Monitor/After/Core/Domain/ServerMonitor.cs
using ServerMonitor.After.Core.Ports;

namespace ServerMonitor.After.Core.Domain
{
    // THE INSIDE (The Core)
    // Pure logic. No references to Console, Twilio, or Kafka.
    public class ServerMonitor
    {
        private readonly IAlertPort _alertPort;

        // We "plug in" the adapter via the constructor (Dependency Injection)
        public ServerMonitor(IAlertPort alertPort)
        {
            _alertPort = alertPort;
        }

        public void CheckTemperature(int temp)
        {
            if (temp > 90)
            {
                // The Core just calls the Port. 
                // It doesn't care if it's an SMS, an Email, or a Carrier Pigeon.
                _alertPort.SendAlert($"Temp is {temp} degrees! Take cover!");
            }
            else
            {
                Console.WriteLine($"[Core] Temp {temp} is normal.");
            }
        }
    }
}

File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/after/core/domain/ServerMonitor.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.core.domain;

import com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.core.ports.AlertPort;
import com.grokkingsoftwarearchitecture.chapter05.shared.LogManager; //Shared Import for cross-cutting concern

public class ServerMonitor {
    private final AlertPort alertPort;

    public ServerMonitor(AlertPort alertPort) {
        this.alertPort = alertPort;
    }

    public void checkTemperature(int temp) {
        if (temp > Constants.HIGH_TEMP_THRESHOLD) {
            alertPort.sendAlert("Temp is " + temp + " degrees! Take cover!");
        } else {
            // Using the shared cross-cutting concern
            LogManager.info(ServerMonitor.class, "[Core] Temp {0} is normal.", temp);
        }
    }
}